stdlib/public/core/Misc.swift

```diff
@@ -102,19 +102,34 @@ func _typeByName(_ name: String) -> Any.Type? {
 @_silgen_name("swift_getTypeByMangledName")
 internal func _getTypeByMangledName(
   _ name: UnsafePointer<UInt8>,
-  _ nameLength: UInt)
+  _ nameLength: UInt,
+  _ numberOfLevels: UInt,
+  _ parametersPerLevel: UnsafePointer<UInt>,
+  _ substitutions: UnsafePointer<Any.Type>)
   -> Any.Type?
 
 /// Lookup a class given a mangled name. This is a placeholder while we bring
 /// up this functionality.
 public  // TEMPORARY
-func _typeByMangledName(_ name: String) -> Any.Type? {
+func _typeByMangledName(_ name: String,
+                        substitutions: [[Any.Type]] = []) -> Any.Type? {
+  // Map the substitutions to a flat representation that's easier to thread
+  // through to the runtime.
+  let numberOfLevels = UInt(substitutions.count)
```

Comment:
Why not have the function take the flat representation instead of performing a representation change?

Reply:
TBH, I'm expecting this testing-only function to get completely removed in the near term, so I don't care much about the API. We'll get to test this code path through the various other adopters soon-ish.